### PR TITLE
Add page titles

### LINKED ISSUE #438: The title of every page is "Pulumi Documentation"
Also, when emitting the title on the page, we always emit the `<h1>{{ page.title }}</h1>` even if the page doesn't have a title (some pages don't, like the CLI documentation).

### PR DESCRIPTION
**Pull request by justinvp**
Use the page's title for the document title. Also, if the page has no title, don't emit the `<h1></h1>` on the page.

Fixes #438

### CODE REVIEW

File: index.md [hunk 1]
```diff
@@ -1,5 +1,4 @@
 ---
-title: Welcome to Pulumi
```
Comment (justinvp) [inline index.md]: This way, the home page's title will be "Pulumi" instead of "Welcome to Pulumi - Pulumi".

Comment (joeduffy) [reply index.md]: We should probably match the www site for title here. This is very important to nail for SEO.

Comment (justinvp) [reply index.md]: Yes, this does match the www title here. It's just "Pulumi".

**Comment by justinvp** [comment]
> Do we need to do a pass over content files and confirm we are setting title in our frontmatter?

If I recall correctly, the last time I looked (as part of the search work) we were setting the title everywhere except the CLI command docs (as those markdown files are generated without the frontmatter) and some manual redirect pages where it won't matter for this. But I'll do another pass.

**Pull request merged**